proto/rill/runtime/v1/queries.proto

```diff
@@ -635,3 +639,14 @@ message TableRowsRequest {
 message TableRowsResponse {
   repeated google.protobuf.Struct data = 1;
 }
+
+message CustomQueryRequest {
+  string instance_id = 1;
+  string sql = 2;
+  int32 priority = 3;
+}
```

Comment:
Let's also support a customizable `uint32 limit` here, with a validation rule that puts a cap in place (maybe 1k rows?)

Reply:
But that way we constraint even local applications

Reply:
btw, with a single limit parameter a user can still overload Runtime (even unintentionally), for example, by joining multiple group by queries:
```sql
SELECT *
FROM
  (SELECT id,
          sum(a)
   FROM a
   GROUP BY id,
            c1
   LIMIT 10000) t1
INNER JOIN
  (SELECT id,
          sum(a)
   FROM a
   GROUP BY id,
            c2
   LIMIT 10000) t2 ON t1.id = t2.id
LIMIT 10000;
```

Reply:
Yeah, that's fine. The goal of putting a `LIMIT` is just to prevent returning huge payloads (which can ruin runtime memory, hit gRPC/ingress limits, or hurt frontend memory/performance). For the actual DuckDB queries, we have CPU and memory limits configured on the DB for each project on cloud, and also have timeouts.